### PR TITLE
realizing the spirit using IPFs

### PR DESCRIPTION
**Pull request by sevu**
A while ago I experimented with the image path functions, especially the new sepia function.
What do you think about it?

(this PR has similar issues like the other one with the variations, let's don't focus on that)

### CODE REVIEW

File: units/Princess.cfg [hunk 1]
```diff
@@ -60,4 +60,157 @@
...
+            {SOUND:HIT_AND_MISS {SOUND_LIST:SWORD_SWISH} {SOUND_LIST:MISS} -75}
+        [/attack_anim]
+        # althought the princess doesn't have an healing ability, this animation is used in a scenario
+        [healing_anim]
+            start_time=0
+            [frame]
+                halo=halo/holy/light-beam-[1~6,7,6~1].png:[30*6,130,70*6]
+            [/frame]
+        [/healing_anim]
+#ifver WESNOTH_VERISON >= 1.13.0
```
Comment (konecnyjakub) [inline units/Princess.cfg]: A typo here. Should be WESNOTH_VERSION.

**Review by konecnyjakub — COMMENTED** [review]
This is interesting. She looks more ghost-like now. I would prefer making the soul a variation as it is very similar to the other type. But maybe we could use inherit=yes to prevent some duplication. Or did you experience any issues with it?

**Comment by sevu** [comment]
Good catch.
I think inherit=yes is not related to the problem with the help pages ( I thought so in the beginning)
It's like, if we have hide_help=yes in the base unit, the variation will show the base unit's help entry.

**Comment by konecnyjakub** [comment]
I just ran Wesnoth and it screamed at me :). (talking about the typo here)

**Comment by sevu** [comment]
Let's add this to milestone 2.0, since it's all about the 1.13 sepia function.

**Comment by konecnyjakub** [comment]
Well, you also made the spirit a variation. If you use inherit=yes for it, it might be worth to have this in 1.3. But if you are not going to do anything with this soon, I will change the milestone.

**Comment by sevu** [comment]
I'll have a look at this next. 
I actually thought about reverting the use of a variation because of the hide_help issue.

**Comment by konecnyjakub** [comment]
Oh, you pushed a new version. Did not notice. I will take a look tomorrow.

**Review by konecnyjakub — COMMENTED** [review]
Looks good overall but there are some small issues. I think that you are using ~BLEND() wrongly. You should use numbers instead of r, g and b. See the last result here: https://github.com/wesnoth/wesnoth/search?l=INI&q=blend&type=&utf8=%E2%9C%93.
Also, the description of variation is identical to the one of the unit type. You can remove it (so it will be inherited).

File: units/Princess.cfg [hunk 2]
```diff
@@ -60,4 +62,118 @@
         [/frame]
         {SOUND:HIT_AND_MISS {SOUND_LIST:SWORD_SWISH} {SOUND_LIST:MISS} -75}
     [/attack_anim]
+
+    [variation]
+        variation_id=spirit
+        inherit=yes
+        variation_name= _ "her soul"
```
Comment (konecnyjakub) [inline units/Princess.cfg]: I am not sure about the variation name. I would use Soul or Spirit (that is without personal pronoun and with first capital letter). But this is not so important.

**Comment by sevu** [comment]
Now I understand that function. About the standing animation, I think the shimmering effect can need more finetuning, but for now I leave it as is.
There are three places where the name & variation_name are used
- the variation_name is shown in the variation line of the base unit. (and looks broken if it isn't set)
- the variation_name is shown under the name in the help page of the variation
- if no "name" is set it won't be shown on the right side when the unit is selected

I think Spirit fit's good.
  
  
  
I will take a look at the other PR soon.

**Comment by konecnyjakub** [comment]
So should I merge this now? If so, please squash the commits.

**Comment by sevu** [comment]
Yes, quashed it. I think there's an option for squash merge too, though I never used it.

Edit: btw, what do you think about the do_not_list attribute? I mostly set it to yes.

**Comment by konecnyjakub** [comment]
The wiki says this:
> Not used by the game, but by tools for browsing and listing the unit tree. If this is 'yes', the unit will be ignored by these tools.

The only such tool I know of is the script used to generate units.wesnoth.org. Given that this unit type (not variation) appears only in a cutscene, it is not important to have it listed there.

**Review by konecnyjakub — APPROVED** [review]
Merging, thank you.

**Comment by sevu** [comment]
My pleasure ;)

**Pull request merged**